src/lib/y2network/connection_config/vlan.rb

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/connection_config/base"
+
+module Y2Network
+  module ConnectionConfig
+    # Configuration for vlan connections
+    class Vlan < Base
+      # @return [Interface, nil]
+      attr_accessor :etherdevice
+      # @return [Integer, nil]
+      attr_accessor :vlan_id
+
+      # Replace the name of the etherdevice by a real Y2Network::Interface
+      #
+      # @see Y2Network::InterfacesCollection
+      #
+      # @param interfaces [Y2Network::InterfacesCollection]
+      def link_related_interfaces(interfaces)
+        return unless etherdevice
+        interface = interfaces.by_name(etherdevice)
+        self.etherdevice = interface if interface
```

Comment:
this is strange construct, as it basically means that etherdevice can be String or Interface dependending when you read it. I do not like much such non-clear contracts of classes.

Reply:
Yep, it is strange. We could have two attributes, one for the name and other for the interface, but also tried to avoid bad use of it although from other point of view give more flexibility.

@jreidinger @imobachgs what do think or propose?

Reply:
We could continue using just interface names and implement proper callbacks when renaming or things like that.

Reply:
etherdevice is defined to be an Interface a few lines above ... so using it in by_name will always fail anyway.

Reply:
It is a problem when reading the configuration. It should be an interface but when reading we have just the name. So, the main discussion was about who and where to put the responsibility of reading.

See https://github.com/yast/yast-network/pull/901/files/7df8eb82498d9206f3df134f8f587447d1919d14#diff-2ac28654c480f56fd97a76755a0dfb15R32

At that point the interface could already exist or not. So we discussed about a two times reading, first use the name in the ifcfg file secondly link to the proper object (at that point all the interfaces should be already declared)

Reply:
@teclator well, my opinion is not have that name here and handle it in reader where it will assign name and in second round assign Interface. All done inside of reader and clearly document that objects are complete only after reader finish.

Reply:
@jreidinger @imobachgs @mchf so, do we prefer to maintain just the names by now for the related interfaces and just the interfaces reader to take care of adding it to the list?

Reply:
i agree with @jreidinger

Reply:
Ok, so will commit some changes and ask for opinions. Thnx both!